ludwig/schema/utils.py

```diff
@@ -16,6 +16,37 @@ def load_config(cls, **kwargs):
     return schema.load(kwargs)
 
 
+def load_trainer_with_kwargs(model_type: str, backend: "Backend", kwargs):  # noqa: F821
+    """Special case of `load_config_with_kwargs` for the trainer schemas.
+
+    In particular, it chooses the correct default type for an incoming config (if it doesn't have one already), but
+    otherwise passes all other parameters through without change.
+    """
+    from ludwig.constants import TYPE
+    from ludwig.trainers.registry import ray_trainers_registry, trainers_registry
+    from ludwig.utils.misc_utils import get_default_from_registry, get_from_registry
+
+    def is_ray():
+        """Check if the backend is ray."""
+        try:
+            from ludwig.backend.ray import RayBackend
+
+            return isinstance(backend, RayBackend)
+        except ImportError:
+            return False
+
+    # Get the default trainer registered for the given model type and backend:
+    trainers_for_model_type = get_from_registry(model_type, ray_trainers_registry if is_ray() else trainers_registry)
```

Comment:
Yes, okay this is much cleaner :) Totally missed that the trainer registries already included model type information lol.

Reply:
changed the logic to infer the trainer config class purely based on model type